app/forms/publish_interface/about_your_organisation_form.rb

```diff
@@ -0,0 +1,89 @@
+module PublishInterface
+  class AboutYourOrganisationForm
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    FIELDS = %i[
+      train_with_us train_with_disability email telephone urn website ukprn
+      address1 address2 address3 address4 postcode region_code
+    ].freeze
+
+    attr_accessor :provider
+    delegate :recruitment_cycle_year, :provider_code, :provider_name, to: :provider
+    attr_accessor :_accredited_bodies_data
+    attr_accessor(*FIELDS)
+
+    def self.build_from_provider(provider)
+      fields_to_populate = provider.attributes.symbolize_keys.slice(*FIELDS)
+
+      form = new(fields_to_populate)
+      form.provider = provider
+      form._accredited_bodies_data = provider.accredited_bodies
+      form
+    end
+
+    def self.build_from_controller_params(params)
+      accredited_bodies_params = params.delete(:accredited_bodies)
+      provider = params.delete(:provider)
+
+      form = new(params)
+      form.provider = provider
+      form._accredited_bodies_data = accredited_bodies_params
+      form.provider.assign_attributes(params)
+      form
+    end
+
+    def accredited_bodies
+      _accredited_bodies_data.map do |ab|
+        AccreditedBody.new(
+          provider_name: ab[:provider_name],
+          provider_code: ab[:provider_code],
+          description: ab[:description],
+        )
+      end
+    end
+
+    def valid?
+      @provider.valid?
+    end
+
+    def save
+      Provider.transaction do
+        update_provider
+        update_accrediting_enrichment
+      end
+      promote_errors
+    end
```

Comment:
```suggestion
    def save
      update_accrediting_enrichment
      if valid?
        Provider.transaction do
         @provider.save
        end
     end
   end
```

 then you can drop ` @provider.save` in `update_accrediting_enrichment`

`promote_errors` feeling a little misplaced as generally `model.save` & `model.valid?` sets the `model.error`

Suggest that we move it to `valid?` to mimic expectation